x-pack/apm-server/main.go

```diff
@@ -100,85 +90,19 @@ type processor interface {
 // newProcessors returns a list of processors which will process
 // events in sequential order, prior to the events being published.
 func newProcessors(args beater.ServerParams) ([]namedProcessor, error) {
-	processors := make([]namedProcessor, 0, 5)
-	const txName = "transaction metrics aggregation"
-	args.Logger.Infof("creating %s with config: %+v", txName, args.Config.Aggregation.Transactions)
-	agg, err := txmetrics.NewAggregator(txmetrics.AggregatorConfig{
-		BatchProcessor:                 args.BatchProcessor,
-		MaxTransactionGroups:           args.Config.Aggregation.Transactions.MaxTransactionGroups,
-		MetricsInterval:                metricsInterval,
-		RollUpIntervals:                rollUpMetricsIntervals,
-		MaxTransactionGroupsPerService: int(math.Ceil(0.1 * float64(args.Config.Aggregation.Transactions.MaxTransactionGroups))),
-		MaxServices:                    args.Config.Aggregation.Transactions.MaxServices,
-		HDRHistogramSignificantFigures: args.Config.Aggregation.Transactions.HDRHistogramSignificantFigures,
-	})
+	processors := make([]namedProcessor, 0, 2)
+
+	name := "LSM aggregator"
+	agg, err := aggregation.NewAggregator(context.TODO(),
+		args.Config.Aggregation.MaxServices,
+		args.Config.Aggregation.Transactions.MaxGroups,
+		args.Config.Aggregation.ServiceTransactions.MaxGroups,
+		args.Config.Aggregation.ServiceDestinations.MaxGroups,
```

Comment:
Service destination still has a fixed default of 10k max groups. It does not change according to memory available. Do we want to change that?

Reply:
I think we should minimise the number of changes. Let's aim to change the technical implementation details (i.e. memory -> LSM) and make improvements in a following release, so long as it's not a one way door.